### PR TITLE
fabricpath adjacency parser and test created - lost PR #60

### PR DESCRIPTION
**Pull request by psolarcz**
Modified as per your comments from #60 

{'domain': {'default': {'interfaces': {'Ethernet2/1': {'hold_time': '00:00:29',
                                                       'level': 1,
                                                       'snpa': 'N/A',
                                                       'state': 'UP',
                                                       'system_id': 'nx-osv-1'},
                                       'Ethernet2/2': {'hold_time': '00:00:27',
                                                       'level': 1,
                                                       'snpa': 'N/A',
                                                       'state': 'UP',
                                                       'system_id': 'nx-osv-2'},
                                       'Ethernet2/4': {'hold_time': '00:00:24',
                                                       'level': 1,
                                                       'snpa': 'N/A',
                                                       'state': 'UP',
                                                       'system_id': 'nx-osv-1'}}}}}

### CODE REVIEW

File: src/genie/libs/parser/nxos/show_fabricpath.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    schema = {
+        Any():
+            {'adj-hold-time-out': str,
+            'adj-intf-name-out': str,
+            'adj-sys-name-out': str,
+             'adj-state-out':str}
+    }
+
+# Python (this imports the Python re module for RegEx)
+import re
```
Comment (felipesasso) [inline src/genie/libs/parser/nxos/show_fabricpath.py]: Please move this to top of file.

File: src/genie/libs/parser/nxos/show_fabricpath.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        p1 = re.compile(r'(?P<peer>(\S+)) + (?P<snpa>(\S+)) + (?P<level>(\d+)) +(?P<state>(UP|DOWN)) + (?P<time>(\S+)) + (?P<interface>(\S+))$')
+
+        # Defines the "for" loop, to pattern match each line of output
+
+        for line in out.splitlines():
+            line = line.strip()
+
+            # Processes the matched patterns for the second line of output
+            m = p1.match(line)
```
Comment (felipesasso) [inline src/genie/libs/parser/nxos/show_fabricpath.py]: Please add output sample here.

File: src/genie/libs/parser/nxos/show_fabricpath.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        p1 = re.compile(r'Sessions +for +VRF +(?P<vrf>(\S+)),'
+                        ' +total: +(?P<total>(\d+)),'
+                        ' +established: +(?P<established>(\d+))$')
+
+
+        # Defines the regex for the next line of device output, which is:
+        # System ID       SNPA            Level  State  Hold Time  Interface
+        # Switch-A           N/A             1      UP     00:00:28   port-channel1
+
+        p1 = re.compile(r'(?P<peer>(\S+)) + (?P<snpa>(\S+)) + (?P<level>(\d+)) +(?P<state>(UP|DOWN)) + (?P<time>(\S+)) + (?P<interface>(\S+))$')
```
Comment (felipesasso) [inline src/genie/libs/parser/nxos/show_fabricpath.py]: Two p1 vars?

File: src/genie/libs/parser/nxos/tests/test_show_fabricpath.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+# Metaparser
+from genie.metaparser.util.exceptions import SchemaEmptyParserError, SchemaMissingKeyError
+
+# nxos show_fabricpath
+from genie.libs.parser.nxos.show_fabricpath import ShowFabricpathIsisAdjacency
+
+# =================================
+# Unit test for 'show fabricpath isis adjacency'
+# =================================
+class test_show_fabricpath_isis_adjacency(unittest.TestCase):
```
Comment (felipesasso) [inline src/genie/libs/parser/nxos/tests/test_show_fabricpath.py]: This should be TestShowFabricpathIsisAdjacency.

File: src/genie/libs/parser/nxos/tests/test_show_fabricpath.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    golden_output1 = {'execute.return_value': '''
+    204-MSMR#show fabricpath isis adjacency
+    Fabricpath IS-IS domain: default Fabricpath IS-IS adjacency database:
+    System ID       SNPA            Level  State  Hold Time  Interface
+    Switch-A        N/A             1      UP     00:00:28   port-channel1
+    Switch-B        N/A             1      UP     00:00:32   port-channel326
+                '''}
+
+    def test_show_fabricpath_isis_adjacency_full1(self):
+        self.maxDiff = None
```
Comment (felipesasso) [inline src/genie/libs/parser/nxos/tests/test_show_fabricpath.py]: We can move this to class level.

File: src/genie/libs/parser/nxos/tests/test_show_fabricpath.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        self.device = Mock(**self.golden_output1)
+        obj = ShowFabricpathIsisAdjacency(device=self.device)
+        parsed_output = obj.parse()
+        self.assertEqual(parsed_output, self.golden_parsed_output1)
+
+    def test_show_fabricpath_isis_adjacency_empty(self):
+        self.maxDiff = None
```
Comment (felipesasso) [inline src/genie/libs/parser/nxos/tests/test_show_fabricpath.py]: This one as well.

File: src/genie/libs/parser/nxos/show_fabricpath.py (hunk 1 above)
Comment (psolarcz) [reply src/genie/libs/parser/nxos/show_fabricpath.py]: Please check commit: 4b4bfa2c51ea119eeef830de97bb979a3c74c800

File: src/genie/libs/parser/nxos/tests/test_show_fabricpath.py (hunk 4 above)
Comment (psolarcz) [reply src/genie/libs/parser/nxos/tests/test_show_fabricpath.py]: Based on your example : 

https://pubhub.devnetcloud.com/media/pyats-development-guide/docs/writeparser/writeparser.html#test-your-new-parser

Comment (tahigash) [reply src/genie/libs/parser/nxos/tests/test_show_fabricpath.py]: Our team is now update all the class names in unittests to follow PEP8. We will update the name later. and we will update the doc as well.

**Pull request merged**